File: src/model/ProductCategoryEnum.ts
export enum ProductCategoryEnum {
  LESSON = 'Aulas',
  TICKETS = 'Ingressos',
  OTHER = 'Outros'
}

export const ProductCategoryList: {
    key: string;
    value: string;
}[] = Object.entries(ProductCategoryEnum).map(([key, value]) => ({ key, value }))

export type ProductCategoryType = keyof typeof ProductCategoryEnum

export const ProductCategoryEntries = Object.entries(ProductCategoryEnum)

File: src/main/java/br/com/api/enums/ProductCategoryEnum.java
package br.com.api.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum ProductCategoryEnum {

	LESSON("label.lesson"), TICKETS("label.tickets"), OTHER("label.other");

	private final String i18n;
}
